pika/adapters/select_connection.py

```diff
@@ -164,6 +163,16 @@ def __init__(self):
         # collection of canceled timeouts
         self._num_cancellations = 0
 
+    def close(self):
```

Comment:
Should all of the connection adapters implement `close`? It looks like the Twisted connection `IOLoopReactorAdapter` does not, and the [Tornado docs](http://www.tornadoweb.org/en/stable/ioloop.html#tornado.ioloop.IOLoop.close) state that `stop` must be called before close which isn't enforced in our code. There's also a good chance I'm mis-understanding how these adapters work

Reply:
@lukebakken - good points. I will follow up in a couple of days. Thanks!

Reply:
@lukebakken, the pull request is ready for resuming code review. Thanks!

Reply:
I also discovered the existing issue #970 in the process of implementing new tests.

Reply:
Great! I'll get on it today.